### PR TITLE
feat: add option to show tabs on the side

### PR DESCRIPTION
**Pull request by satya164**
**Motivation**

This adds ability to render the tab bar on left/right as a sidebar. It useful on larger screens where we may want to take advantage of the available horizontal screen and show a sidebar instead of a bottom tab bar.

**Test plan**

Open the "Bottom Tabs" example and resize the screen to see the different modes of the tab bar. Tap on the "<<" button on the bottom left to switch between a smaller and larger sidebar.

**Video**

https://github.com/react-navigation/react-navigation/assets/1174278/224beaa9-8e26-4d94-a70d-11e54c37d944

### CODE REVIEW

File: packages/bottom-tabs/src/views/TabBarIcon.tsx [hunk 1]
```diff
@@ -50,25 +50,21 @@ export function TabBarIcon({
...
       <Badge
         visible={badge != null}
-        style={[
-          styles.badge,
-          horizontal ? styles.badgeHorizontal : styles.badgeVertical,
-          badgeStyle,
-        ]}
-        size={(size * 3) / 4}
+        style={[styles.badge, badgeStyle]}
+        size={(ICON_SIZE * 3) / 4}
```
Comment (osdnk) [inline packages/bottom-tabs/src/views/TabBarIcon.tsx:67]: * 0.75?

File: packages/bottom-tabs/src/views/BottomTabBar.tsx [hunk 2]
```diff
@@ -256,42 +270,67 @@ export function BottomTabBar({
...
+                          StyleSheet.hairlineWidth,
+                        0,
+                      ],
+                    }),
+                  },
                 ],
-              }),
+                // Absolutely position the tab bar so that the content is below it
+                // This is needed to avoid gap at bottom when the tab bar is hidden
+                position: isTabBarHidden ? 'absolute' : (null as any),
```
Comment (osdnk) [inline packages/bottom-tabs/src/views/BottomTabBar.tsx:301]: Hm, I'm not sure how you like this pattern, but I typically use

```
   ...(isTabBarHidden && { position: 'hidden' })
```

**Review by osdnk — COMMENTED** [review]
The code works and looks good in general. One question is if the name "bottom-tab" is still valid?

File: packages/bottom-tabs/src/views/BottomTabBar.tsx (hunk 2 above)
Comment (satya164) [reply packages/bottom-tabs/src/views/BottomTabBar.tsx:301]: Changed to `position: isTabBarHidden ? 'absolute' : undefined`

**Comment by satya164** [comment]
> @osdnk: One question is if the name "bottom-tab" is still valid

You could make the argument is that the default one is still bottom, and you could position tabs at the top in material-top-tabs so it's like this already 😅

But yeah, the only real argument I have is to avoid another big breaking change (especially to keep older blog posts, examples etc. working) and to keep our download counts on npm.

**Pull request merged**